scripts/exregional_run_prdgen.sh

```diff
@@ -0,0 +1,437 @@
+
+else
+  echo "this grid is not ready for parallel prdgen: ${PREDEF_GRID_NAME}"
+fi
+else
+#
+# use single core to process all addition grids.
+#
+if [ ${#ADDNL_OUTPUT_GRIDS[@]} -gt 0 ]; then
+
+  cd_vrfy ${COMOUT}
+
+  grid_specs_130="lambert:265:25.000000 233.862000:451:13545.000000 16.281000:337:13545.000000"
+  grid_specs_200="lambert:253:50.000000 285.720000:108:16232.000000 16.201000:94:16232.000000"
+  grid_specs_221="lambert:253:50.000000 214.500000:349:32463.000000 1.000000:277:32463.000000"
+  grid_specs_242="nps:225:60.000000 187.000000:553:11250.000000 30.000000:425:11250.000000"
+  grid_specs_243="latlon 190.0:126:0.400 10.000:101:0.400"
+  grid_specs_clue="lambert:262.5:38.5 239.891:1620:3000.0 20.971:1120:3000.0"
+  grid_specs_hrrr="lambert:-97.5:38.5 -122.719528:1799:3000.0 21.138123:1059:3000.0"
+  grid_specs_hrrre="lambert:-97.5:38.5 -122.719528:1800:3000.0 21.138123:1060:3000.0"
+  grid_specs_rrfsak="lambert:-161.5:63.0 172.102615:1379:3000.0 45.84576:1003:3000.0"
+  grid_specs_hrrrak="nps:225:60.000000 185.117126:1299:3000.0 41.612949:919:3000.0"
+
+  for grid in ${ADDNL_OUTPUT_GRIDS[@]}
+  do
+    for leveltype in prslev natlev ififip testbed
+    do
+      
+      eval grid_specs=\$grid_specs_${grid}
+      subdir=${COMOUT}/${grid}_grid
+      mkdir -p ${subdir}/${fhr}
+      bg_remap=${subdir}/${NET}.${cycle}${dot_ensmem}.${leveltype}.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+
+      # Interpolate fields to new grid
+      eval infile=\${leveltype}
+      if [ ${NET} = "RRFS_NA_13km" ]; then
+         wgrib2 ${infile} -set_bitmap 1 -set_grib_type c3 -new_grid_winds grid \
+           -new_grid_vectors "UGRD:VGRD:USTM:VSTM:VUCSH:VVCSH" \
+           -new_grid_interpolation bilinear \
+           -if ":(WEASD|APCP|NCPCP|ACPCP|SNOD):" -new_grid_interpolation budget -fi \
+           -if ":(NCONCD|NCCICE|SPNCR|CLWMR|CICE|RWMR|SNMR|GRLE|PMTF|PMTC|REFC|CSNOW|CICEP|CFRZR|CRAIN|LAND|ICEC|TMP:surface|VEG|CCOND|SFEXC|MSLMA|PRES:tropopause|LAI|HPBL|HGT:planetary boundary layer):|ICPRB|SIPD|ICSEV" -new_grid_interpolation neighbor -fi \
+           -new_grid ${grid_specs} ${subdir}/${fhr}/tmp_${grid}.grib2 &
+      else
+         wgrib2 ${infile} -set_bitmap 1 -set_grib_type c3 -new_grid_winds grid \
+           -new_grid_vectors "UGRD:VGRD:USTM:VSTM:VUCSH:VVCSH" \
+           -new_grid_interpolation neighbor \
+           -new_grid ${grid_specs} ${subdir}/${fhr}/tmp_${grid}.grib2 &
+      fi
+      wait 
+
+      # Merge vector field records
+      wgrib2 ${subdir}/${fhr}/tmp_${grid}.grib2 -new_grid_vectors "UGRD:VGRD:USTM:VSTM:VUCSH:VVCSH" -submsg_uv ${bg_remap} &
+      wait 
+
+      # Remove temporary files
+      rm -f ${subdir}/${fhr}/tmp_${grid}.grib2
+
+      # Save to com directory 
+      mkdir -p ${COMOUT}/${grid}_grid
+      cp_vrfy ${bg_remap} ${COMOUT}/${grid}_grid/${NET}.${cycle}${dot_ensmem}.${leveltype}.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+
+      if [ $leveltype = 'prslev' ]; then
```

Comment:
I may have overlooked it, but each of these if statements contain roughly the same information. Would this one command work to replace them all? I've replaced the hardcoded `prslev` with `${leveltype}` instead.

```
         ln_vrfy -fs --relative ${COMOUT}/${grid}_grid/${NET}.${cycle}${dot_ensmem}.${leveltype}.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2 ${COMOUT}/${net4}.${cycle}.${leveltype}.f${fhr}.${gridname}grib2
         wgrib2 ${COMOUT}/${net4}.${cycle}.prslev.f${fhr}.${gridname}grib2 -s > ${COMOUT}/${net4}.${cycle}.${leveltype}.f${fhr}.${gridname}grib2.idx
```

Reply:
Yes I agree.  I'm about to test a new version of the script with this change and the ones you recommended above with adding loops.  Once that's done I'll commit the changes and mark these comments as resolved.